### ISSUE TITLE
Setup new TOTP without disabling

### ISSUE THREAD

**Issue by rullzer** [nextcloud/twofactor_totp#678]
I just had to move all my TOTP codes to my new phone.
However in order to do this I have to disable and re-enable the TOTP setting.

While valid it does feel a little... counter intuitive.

I'd prefer a button 'setup new TOTP' or whatever that guides us trough the wizard again (also warning previous codes are invalid). Would feel a bit more user friendly IMO.

**Comment by isdnfan** [comment]
I would appreciate such an option as well. unfortunately same request was rejected some time ago #158 
At least for limited number of devices, say max 5.

**Comment by ChristophWurst** [comment]
@isdnfan re-read @rullzer's suggestions. This isn't about allowing more than one simultaneous code, it's about a simpler UX flow. With this approach the old registrations will still be invalidated.

**Comment by isdnfan** [comment]
@ChristophWurst I agree the request isn't exact the same. From the wording 'setup new TOTP' I understood what I looked for.. 

In general only one TOTP code is not ideal - the user can't pair multiple devices - like phone and tablet - for TOTP (or has to pair them at same time). Other platforms like Google and Microsoft allow multiple TOTP devices - Nextcloud with Webauthn as well - why it is impossible to have multiple TOTP identified by friendly device name which could be invalidated one by one once the user stops using specific device?

**Comment by ChristophWurst** [comment]
> Other platforms like Google and Microsoft allow multiple TOTP devices

Proof? At least for Google I find official and unofficial sources that say you need to reset TOTP and scan the QR code with all your devices at once. Like exactly how you can set up more than one device here.

**Comment by isdnfan** [comment]
here a screenshot from MS O365 security page: 3 different authenticator apps are registered:
![image](https://user-images.githubusercontent.com/18125597/106584442-792c6200-6546-11eb-82ab-77487491514a.png)

**Comment by ToeiRei** [comment]
I use hardware and an authenticator app as backup in case I left my usb key at home. I would love to have the same way on nextcloud too.

**Comment by ernolf** [comment]
Anyone who needs this feature should switch to twofactor_oath:

https://github.com/ernolf/twofactor_oath

It is built for everything twofactor_totp cannot do: hardware OATH tokens, configurable algorithm and length, predetermined (custom) secrets, challenge-response, admin enrollment for whole groups, and mass provisioning. The simple cases stay simple, the advanced cases become possible.

h.t.h.

ernolf
